wemix/governance-contract/common/simulated-backend/helper.go

```diff
@@ -0,0 +1,74 @@
+package sim
+
+import (
+	"encoding/json"
+	"strings"
+	"testing"
+
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/core/vm"
+	"github.com/stretchr/testify/require"
+)
+
+func parseABI(abiDefinition interface{}) (*abi.ABI, error) {
+	s, ok := abiDefinition.(string)
+	if !ok {
+		if bytes, err := json.Marshal(abiDefinition); err != nil {
+			return nil, err
+		} else {
+			s = string(bytes)
+		}
+	}
+	if abi, err := abi.JSON(strings.NewReader(s)); err != nil {
+		return nil, err
+	} else {
+		return &abi, nil
+	}
+}
+
+func StringToBytes32(str string) [32]byte {
+	dest := [32]byte{}
+	copy(dest[:], []byte(str))
+	return dest
```

Comment:
requires processing for sizes greater than 32.
```suggestion
	bytes := []byte(str)
	if len(bytes) > 32 {
		return bytes[:32]
	}
	copied := make([]byte, 32)
	copy(copied[32-len(bytes):32], bytes)
	return copied
```